### PR TITLE
feat(query): support settings admin api for global level settings

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

feat(query): support settings admin api for global level settings

```shell
curl -XGET 'http://127.0.0.1:8080/v1/tenants/test_tenant/settings'    
curl -XPOST 'http://127.0.0.1:8080/v1/tenants/test_tenant/settings/max_threads' -H "Content-Type: application/json" --data "\"2\""
curl -XDELETE 'http://127.0.0.1:8080/v1/tenants/test_tenant/settings/max_threads'
```

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14901)

### CODE REVIEW

File: src/query/service/src/api/http/v1/settings.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+pub struct SettingsItem {
+    pub name: String,
+    pub desc: &'static str,
+    pub user_value: String,
+    pub default_value: String,
+    pub range: Option<String>,
+}
+
+async fn list_settings_impl(tenant: &str) -> Result<Vec<SettingsItem>> {
+    if tenant.is_empty() {
```
Comment (TCeason) [inline src/query/service/src/api/http/v1/settings.rs:33]: NonEmptyString::new will check empty.

File: src/query/service/src/api/http/v1/settings.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            desc: item.desc,
+            user_value: item.user_value.to_string(),
+            default_value: item.default_value.to_string(),
+            range: item.range.map(|x| x.to_string()),
+        })
+        .collect::<Vec<_>>())
+}
+
+async fn set_setting_impl(tenant: &str, key: &str, value: String) -> Result<Vec<SettingsItem>> {
+    if tenant.is_empty() {
```
Comment (TCeason) [inline src/query/service/src/api/http/v1/settings.rs:55]: Ditto

File: src/query/service/src/api/http/v1/settings.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            desc: item.desc,
+            user_value: item.user_value.to_string(),
+            default_value: item.default_value.to_string(),
+            range: item.range.map(|x| x.to_string()),
+        })
+        .collect::<Vec<_>>())
+}
+
+async fn unset_setting_impl(tenant: &str, key: &str) -> Result<Vec<SettingsItem>> {
+    if tenant.is_empty() {
```
Comment (TCeason) [inline src/query/service/src/api/http/v1/settings.rs:83]: Ditto

File: src/query/service/src/api/http/v1/settings.rs (hunk 1 above)
Comment (zhang2014) [reply src/query/service/src/api/http/v1/settings.rs:33]: More detail error message.

**Pull request merged**